datafusion/sqllogictest/test_files/union_function.slt

```diff
@@ -23,7 +26,8 @@ query ?I
 select union_column, union_extract(union_column, 'int') from union_table;
 ----
 {int=1} 1
-{int=2} 2
+{string=bar} NULL
```

Comment:
I added a new row to the table so we could test union_tag with a field that did not exist, per @Omega359 's suggestion